llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -1670,8 +1670,12 @@ bool VectorCombine::foldShuffleOfShuffles(Instruction &I) {
 
 using InstLane = std::pair<Value *, int>;
 
-static InstLane lookThroughShuffles(Value *V, int Lane) {
+static InstLane
+lookThroughShuffles(Value *V, int Lane,
+                    SmallPtrSetImpl<Instruction *> *VisitedShuffles) {
```

Comment:
`*VisitedShuffles = nullptr` in signature, so you don't have to pass nullptr explicitly?